src/librustdoc/html/render/write_shared.rs

```diff
@@ -85,106 +180,66 @@ pub(super) fn write_shared(
 
         // Handle the official themes
         match theme {
-            "light" => write_minify(
-                &cx.shared.fs,
-                cx.path("light.css"),
-                static_files::themes::LIGHT,
-                options.enable_minification,
-            )?,
-            "dark" => write_minify(
-                &cx.shared.fs,
-                cx.path("dark.css"),
-                static_files::themes::DARK,
-                options.enable_minification,
-            )?,
-            "ayu" => write_minify(
-                &cx.shared.fs,
-                cx.path("ayu.css"),
-                static_files::themes::AYU,
-                options.enable_minification,
-            )?,
+            "light" => write_minify("light.css", static_files::themes::LIGHT)?,
+            "dark" => write_minify("dark.css", static_files::themes::DARK)?,
+            "ayu" => write_minify("ayu.css", static_files::themes::AYU)?,
             _ => {
                 // Handle added third-party themes
                 let content = try_err!(fs::read(&entry.path), &entry.path);
-                cx.shared
-                    .fs
-                    .write(cx.path(&format!("{}.{}", theme, extension)), content.as_slice())?;
+                // This is not exactly right: if compiled a second time with the same toolchain but different CLI args, the file could be different.
+                // But docs.rs doesn't use this, so hopefully the issue doesn't come up.
+                write_toolchain(&format!("{}.{}", theme, extension), content.as_slice())?;
```

Comment:
Hmm, now that I think about it I don't think this should be a shared file at all. If someone passes `--theme x.css`, they'll link to `x.css` in their docs, they won't expect a resource suffix.

Reply:
Oh well we already added it before, I don't think that change should be part of this PR. `cx.path` adds the resource suffix.

Reply:
Does `rustdoc` not take care of generating the link to `x.css` itself when it's a custom theme?

Reply:
Oh, you're right. Ignore me. https://github.com/rust-lang/rust/blob/a5029ac0ab372aec515db2e718da6d7787f3d122/src/librustdoc/html/layout.rs#L181-L195